haystack/core/pipeline/base.py

```diff
@@ -308,6 +308,30 @@ def add_component(self, name: str, instance: Component) -> None:
             visits=0,
         )
 
+    def remove_component(self, name: str) -> None:
+        """
+        Remove a component from the pipeline.
+
+        Remove an existing component from the pipeline by providing its name.
+        All edges that connect to the component will also be deleted.
+
+        :param name:
+            The name of the component to remove.
+
+        :raises ValueError:
+            If there is no component with that name already in the Pipeline.
+        """
+
+        # Check that a component with that name is in the Pipeline
+        if name not in self.graph.nodes:
+            raise ValueError(
+                f"There is no component named '{name}' in the pipeline. The valid component names are: ",
+                ", ".join(n for n in self.graph.nodes),
+            )
+
+        # Delete component from the graph, deleting all its connections
+        self.graph.remove_node(name)
```

Comment:
This is not enough. The Component instance holds some connection info, the Component instance would become unusable if that is not removed.

Reply:
Okey, handeling that riht now.

Reply:
I believe I solved it.

Reply:
There's some more work required, I'll handle it.